xbmc/video/windows/GUIWindowVideoBase.cpp

```diff
@@ -897,10 +897,12 @@ void CGUIWindowVideoBase::GetContextButtons(int itemNumber, CContextButtons &but
         buttons.Add(CONTEXT_BUTTON_PLAY_PARTYMODE, 15216); // Play in Partymode
       }
 
-      //if the item isn't a folder or script, is a member of a list rather than a single item
-      //and we're not on the last element of the list,
-      //then add add either 'play from here' or 'play only this' depending on default behaviour
-      if (!(item->m_bIsFolder || item->IsScript()) && m_vecItems->Size() > 1 && itemNumber < m_vecItems->Size()-1)
+      // if the item isn't a folder or script, is not explicitly marked as not playable,
+      // is a member of a list rather than a single item and we're not on the last element of the list,
+      // then add either 'play from here' or 'play only this' depending on default behaviour
+      if (!(item->m_bIsFolder || item->IsScript()) &&
+          (!item->HasProperty("IsPlayable") || item->GetProperty("IsPlayable").asBoolean()) &&
+          m_vecItems->Size() > 1 && itemNumber < m_vecItems->Size() - 1)
       {
```

Comment:
The likelihood of causing regressions here is still quite big here. Maybe do the following condition (although completely ugly...)?

```cpp
if ((!(item->m_bIsFolder || item->IsScript()) &&
          m_vecItems->Size() > 1 && itemNumber < m_vecItems->Size() - 1 &&
           (!item->HasProperty("IsPlayable") || item->GetProperty("IsPlayable").asBoolean())) || (item->HasProperty("IsPlayable") && !item->GetProperty("IsPlayable").asBoolean()))
```

Condition is ugly but at least there is a way to disable the context menu item from python without causing regressions

Reply:
`(item->HasProperty("IsPlayable") && !item->GetProperty("IsPlayable").asBoolean()))` is the same as `(!item->HasProperty("IsPlayable") || item->GetProperty("IsPlayable").asBoolean()))` negated... I think we don't want that

Reply:
Yeah, keep forgetting this is to show the contextmenu item and not to hide it. Updated the example condition

Reply:
disregard the comments, brainfart on my side. Original condition is okay.